gherkin/billing_charges_test.go

```diff
@@ -0,0 +1,585 @@
+			if err != nil {
+				panic(err)
+			}
+			sql := string(content)
+			rows, err := db.Query(sql)
+			if err != nil {
+				panic(err)
+			}
+			_ = rows
+		}
+
+		// defer db.Close()
+	})
+
+	ctx.AfterSuite(func() { fmt.Println("After running test suite") })
+}
+
+func InitializeScenario(ctx *godog.ScenarioContext) {
+	// Background
+	ctx.Step(`^clear out data from database tables: ([A-Za-z_ ,]+)$`, clearDatabaseTables)
+	ctx.Step(`^a clean billing database$`, aCleanBillingDatabase)
+
+	// Given
+	ctx.Step(`^(?:a|the) tenant has a ([A-Za-z_\- \.0-9]+) between \'(\d+)-(\d+)-(\d+) (\d+):(\d+):(\d+)\' and \'(\d+)-(\d+)-(\d+) (\d+):(\d+):(\d+)\'$`, aTenantHasSomethingBetweenyyyymmddHHMMssAndyyyymmddHHMMss)
+	ctx.Step(`^(?:a|the) tenant has a ([A-Za-z_\- \.0-9]+) between \'(\d+)-(\d+)-(\d+) (\d+):(\d+):(\d+)\' and \'(\d+)-(\d+)-(\d+) (\d+):(\d+)\'$`, aTenantHasSomethingBetweenyyyymmddHHMMssAndyyyymmddHHMM)
+	ctx.Step(`^(?:a|the) tenant has a ([A-Za-z_\- \.0-9]+) between \'(\d+)-(\d+)-(\d+) (\d+):(\d+)\' and \'(\d+)-(\d+)-(\d+)\'$`, aTenantHasSomethingBetweenyyyymmddHHMMAndyyyymmdd)
+	ctx.Step(`^(?:a|the) tenant has a ([A-Za-z_\- \.0-9]+) between \'(\d+)-(\d+)-(\d+) (\d+):(\d+)\' and \'(\d+)-(\d+)-(\d+) (\d+):(\d+)\'$`, aTenantHasSomethingBetweenyyyymmddHHMMAndyyyymmddHHMM)
+	ctx.Step(`^(?:a|the) tenant has a ([A-Za-z_\- \.0-9]+) between \'(\d+)-(\d+)-(\d+)\' and \'(\d+)-(\d+)-(\d+)\'$`, aTenantHasSomethingBetweenyyyymmddAndyyyymmdd)
+
+	// When
+	ctx.Step(`^billing is run for ([A-Za-z 0-9]+)$`, billingIsRun)
+
+	// Then
+	ctx.Step(`^the bill, including VAT, should be £(\d+)\.(\d+)$`, theBillShouldBe)
+}
+
+// Background
+
+// Assumes tables being passed in as a comma-separated list. This is in case we want to use this function directly from Gherkin.
+func clearDatabaseTables(tables string) error {
+
+	fmt.Print("\n\n#################################################################\n")
+	fmt.Printf("\nClearing out any existing data from tables populated in the tests (%s).\n\n", tables)
+
+	tables = strings.Replace(tables, " ", "", -1)
+	tableList := strings.Split(tables, ",")
+
+	for i := 0; i < len(tableList); i++ {
+		sql := fmt.Sprintf("DELETE FROM %s;", tableList[i])
+		// fmt.Printf("Running '%s'\n", sql)
+		rows, err := db.Query(sql)
+		if err != nil {
+			panic(err)
+		}
+		_ = rows
+	}
+
+	return nil
+}
+
+func aCleanBillingDatabase() error {
+	// Clear out any data from previous tests.
+	tableList := "compose_audit_events, consolidated_billable_events, consolidation_history, events, app_usage_events, service_usage_events, currency_rates, vat_rates, pricing_plans, pricing_plan_components"
+	clearDatabaseTables(tableList)
+
+	// Add data to the following tables:
+	// vat_rates, currency_rates, pricing_plans, pricing_plan_components.
+
+	planConfig, err := eventstore.LoadConfig("../config.json")
+	if err != nil {
+	  panic(err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 10 * time.Minute )
```

Comment:
I believe unless we're planning to use a different context, the background one makes more sense
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute )
```